src/components/ha-form/ha-form-select.ts

```diff
@@ -26,7 +28,36 @@ export class HaFormSelect extends LitElement implements HaFormElement {
 
   protected render(): TemplateResult {
     return html`
-      <ha-paper-dropdown-menu .label=${this.label}>
+      <paper-menu-button horizontal-align="right" vertical-offset="8">
+        <div class="dropdown-trigger" slot="dropdown-trigger">
+          <paper-ripple></paper-ripple>
+          <paper-input
+            id="input"
+            type="text"
+            readonly
+            value=${this.data}
+            label=${this.label}
+            input-role="button"
+            input-aria-haspopup="listbox"
+            autocomplete="off"
+          >
+            ${this.data &&
+            (this.schema.optional !== undefined
+              ? this.schema.optional
+              : !this.schema.required)
```

Comment:
This is not needed, either `optional` or `required` is true
```suggestion
            this.schema.optional
```